app/controllers/api/v1/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+        include ErrorRenderable
+
+        rescue_from ArgumentError, with: :render_bad_request
+
+        def google_oauth2
+          @user = User.from_omniauth(auth)
+
+          if @user.persisted?
+            render_success
+          else
+            render_errors(
+              details: @user.errors.full_messages,
+              status: :unprocessable_entity
+            )
+          end
```

Comment:
🎨 Just to reduce the nested code level

```suggestion
          return render_success if @user.persisted?
           
          render_errors(
            details: @user.errors.full_messages,
            status: :unprocessable_entity
          )
```